src/archive/Archive.zig

```diff
@@ -587,111 +586,106 @@ pub fn parse(self: *Archive, allocator: *Allocator, stderr: anytype) !void {
         while (true) {
             var first_line_buffer: [gnu_first_line_buffer_length]u8 = undefined;
 
-            const has_line_to_process = result: {
-                const chars_read = reader.read(&first_line_buffer) catch |err| switch (err) {
-                    else => |e| return e,
-                };
-
-                if (chars_read < first_line_buffer.len) {
-                    break :result false;
-                }
-
-                break :result true;
+            const chars_read = reader.read(&first_line_buffer) catch |err| switch (err) {
+                else => |e| return e,
             };
 
```

Comment:
```suggestion
if (chars_read == 0) {
    return;
}
```

An archive containing only the magic is still valid, we cant error on that. This case need to be specially handled.